prow/plugins/helpers.go

```
@@ -0,0 +1,106 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugins
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// Apply returns a policy that merges the child into the parent
+func Apply(parent, child json.RawMessage) (json.RawMessage, error) {
+	if len(parent) == 0 {
+		return child, nil
+	}
+	var p, c map[string]interface{}
+	json.Unmarshal(parent, &p)
+	json.Unmarshal(child, &c)
+	for k, v := range c {
+		p[k] = v
+	}
+	return json.Marshal(p)
+}
+
+// ConfigTree specifies the global generic config for a plugin.
+type ConfigTree struct {
+	Config json.RawMessage `json:"config,omitempty"`
+	Orgs   map[string]Org  `json:"orgs,omitempty"`
+}
+
+// GetOrg returns the org config after merging in any global config.
+func (t ConfigTree) GetOrg(name string) *Org {
+	o, ok := t.Orgs[name]
+	if ok {
+		o.Config, _ = Apply(t.Config, o.Config)
+	} else {
+		o.Config = t.Config
+	}
+	return &o
+}
+
+// Org holds the default config for an entire org, as well as any repo overrides.
+type Org struct {
+	Config json.RawMessage `json:"config,omitempty"`
+	Repos  map[string]Repo `json:"repos,omitempty"`
+}
+
+// GetRepo returns the repo config after merging in any org policies.
+func (o Org) GetRepo(name string) *Repo {
+	r, ok := o.Repos[name]
+	if ok {
+		r.Config, _ = Apply(o.Config, r.Config)
```

Why ignore the error here?